epam/NGB
# Short URLs generation and handling

# Description

## Background

This PR adds new functionality of generating and handling short URLs instead of long URLs.

## Changes

New methods for UtilController were added:
* generateShortUrl - takes the original url and alias(optional) and generate a short postfix for storing the original url (if alias is specified but isn't unique postfix based on a hash-function will be generated`)
* redirectToOriginalUrlByAlias - takes a short postfix and searches for the original url, if it exists controller will redirect to the original url, otherwise will redirect to 404 error page

New DAO class was added.
Tests for DAO and Manager were provided.

## Acceptance criteria
```
* Test case 1
    * Try to generate with swagger short postfixes for your original NGB-URLs with custom allowed alias and without
    * Go to `{your-NGB-server}/navigate?alias={postfix}` - it should redirect you to the original link
 
* Test case 2
    * Configure your NGB server with the parameter  `short.link.expired.period=0` (it means that short-URLs will immediately expire and be removed by a next `generateShortUrl()` method call)
    * Generate 2 short URLs and check that the first one has already expired. 
```

# Check list

- [x] Unit tests are provided
- [ ] Integration tests are provided (if CLI/API was changed)
